plugins/inputs/consul/consul.go

```diff
@@ -0,0 +1,174 @@
+package consul
+
+import (
+	"net/http"
+	"reflect"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var fieldMap = map[string]string{
+	"check_id":     "CheckID",
+	"check_name":   "Name",
+	"node":         "Node",
+	"service_id":   "ServiceID",
+	"service_name": "ServiceName",
+	"status":       "Status",
+	"notes":        "Notes",
+	"output":       "Output",
+}
+
+type Consul struct {
+	Address    string
+	Scheme     string
+	Token      string
+	Username   string
+	Password   string
+	Datacentre string
+	Fields     []string
```

Comment:
Related to my other comment: It will be easier to add new measurements/endpoints to this input if Fields and Tags could be broken out into endpoint-specific configs. Similar to the way the Jolokia input does metrics: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/jolokia/jolokia.go